Signal/src/ViewControllers/MessagesViewController.m

```diff
@@ -500,6 +524,133 @@ - (void)viewWillAppear:(BOOL)animated
                                                                                                                @"Short name for edit menu item to share contents of media message.")
                                                                                       action:shareSelector],
                                                            ];
+
+    [self ensureBlockStateIndicator];
+}
+
+- (void)setUserHasScrolled:(BOOL)userHasScrolled {
+    _userHasScrolled = userHasScrolled;
+    
+    [self ensureBlockStateIndicator];
+}
+
+- (void)ensureBlockStateIndicator
+{
+    // This method should be called rarely, so it's simplest to discard and
+    // rebuild the indicator view every time.
+    [self.blockStateIndicator removeFromSuperview];
+    self.blockStateIndicator = nil;
+    
+    if (self.userHasScrolled) {
+        return;
+    }
+
+    NSString *blockStateMessage = nil;
+    if ([self isBlockedContactConversation]) {
+        blockStateMessage = NSLocalizedString(@"MESSAGES_VIEW_CONTACT_BLOCKED",
+                                              @"Indicates that this 1:1 conversation has been blocked.");
+    } else {
+        int blockedGroupMemberCount = [self blockedGroupMemberCount];
+        if (blockedGroupMemberCount == 1) {
+            blockStateMessage = [NSString stringWithFormat:NSLocalizedString(@"MESSAGES_VIEW_GROUP_1_MEMBER_BLOCKED_FORMAT",
+                                                                             @"Indicates that a single member of this group has been blocked."),
+                                 blockedGroupMemberCount];
+        } else if (blockedGroupMemberCount > 1) {
+            blockStateMessage = [NSString stringWithFormat:NSLocalizedString(@"MESSAGES_VIEW_GROUP_N_MEMBERS_BLOCKED_FORMAT",
+                                                                             @"Indicates that some members of this group has been blocked. Embeds "
+                                                                             @"{{the number of blocked user in this group}}."),
+                                 blockedGroupMemberCount];
+        }
+    }
+    
+    if (blockStateMessage) {
+        UILabel *label = [UILabel new];
+        label.font = [UIFont ows_mediumFontWithSize:14.f];
+        label.text = blockStateMessage;
+        label.textColor = [UIColor whiteColor];
+        
+        UIView * blockStateIndicator = [UIView new];
+        blockStateIndicator.backgroundColor = [UIColor ows_signalBrandBlueColor];
```

Comment:
I'm thinking we should use a color that is more "alert-y", somewhere on the red, orange, yellow spectrum - to draw the eye and to emphasize that some functionality is disabled.

Here's `[UIColor ows_redColor]`:

![screen shot 2017-04-05 at 11 29 12 am](https://cloud.githubusercontent.com/assets/625803/24713276/2f504ab8-19f3-11e7-87c4-b957593d2490.png)


WDYT?